### PR TITLE
Invert if statement in filesystem for clearer intent.

### PR DESCRIPTION
**Pull request by ghost**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.
From the top down of the code, the code checks to see if the file sizes are different, then to check if they are 0, and then to compare the memory contents. However, the first early return in that portion is to return DifferentContents, and then check for 0 sizes and if they are then return SameContents. However on the bottom, it checks to see if the memory difference is 0.

To make the code clearer in its intent, I changed the if statement so it would return DifferentContents if memcmp was non-0, and then otherwise say they are the same contents.

### CODE REVIEW

File: lib/Basic/FileSystem.cpp [hunk 1]
```diff
@@ -228,10 +228,10 @@ swift::areFilesDifferent(const llvm::Twine &source,
   if (destRegionErr)
     return convertDestinationError(destRegionErr);
 
-  if (0 == memcmp(sourceRegion.const_data(), destRegion.const_data(), size))
-    return FileDifference::SameContents;
+  if (memcmp(sourceRegion.const_data(), destRegion.const_data(), size)))
```
Comment (harlanhaskins) [inline lib/Basic/FileSystem.cpp:231]: Nitpick: Can you explicitly compare this to `!= 0`? Every time I use or read a use of `memcmp` I forget that it returns 0 if the regions are equal, and it helps to keep the explicit comparison to 0

Comment (CodaFi) [reply lib/Basic/FileSystem.cpp:231]: There's also an extra `)` here.

**Comment by CodaFi** [comment]
While you're here, could you explicitly brace all of the blocks in `swift::areFilesDifferent`? It's a pet peeve of mine to see multi-line blocks without them.

**Comment by pi1024e** [comment]
@CodaFi done!

File: lib/Basic/FileSystem.cpp [hunk 2]
```diff
@@ -204,34 +207,39 @@ swift::areFilesDifferent(const llvm::Twine &source,
...
                                     size, 0, destRegionErr);
 
-  if (destRegionErr)
+  if (destRegionErr) {
     return convertDestinationError(destRegionErr);
+  }
 
-  if (0 == memcmp(sourceRegion.const_data(), destRegion.const_data(), size))
-    return FileDifference::SameContents;
+  if (memcmp(sourceRegion.const_data(), destRegion.const_data(), size)) != 0) {
```
Comment (CodaFi) [inline lib/Basic/FileSystem.cpp:238]: There's still an extra `)`

```diff
- if (memcmp(sourceRegion.const_data(), destRegion.const_data(), size)) != 0) {
+ if (memcmp(sourceRegion.const_data(), destRegion.const_data(), size) != 0) {
```

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Review by CodaFi — APPROVED** [review]

**Comment by CodaFi** [comment]
@swift-ci smoke test Linux

**Comment by CodaFi** [comment]
⛵

**Pull request merged**